File: base/src/main/java/com/widget/RoundRectCornerImageView.java
package com.widget;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Canvas;
import android.graphics.Path;
import android.graphics.RectF;
import android.util.AttributeSet;

import com.base.R;

public class RoundRectCornerImageView extends android.support.v7.widget.AppCompatImageView {

    private float radius = 5.0f;
    private Path path;
    private RectF rect;

    public RoundRectCornerImageView(Context context) {
        super(context);
        init(context, null);
    }

    public RoundRectCornerImageView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context, attrs);
    }

    public RoundRectCornerImageView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        init(context, attrs);
    }

    private void init(Context context, AttributeSet attrs) {
        path = new Path();
        rect = new RectF();
        if (attrs != null) {
            TypedArray typedArray = context.obtainStyledAttributes(attrs, R.styleable.RoundRectCornerImageView);
            radius = typedArray.getFloat(R.styleable.RoundRectCornerImageView_radius, 5.0f);
            typedArray.recycle();
        }
    }

    @Override
    protected void onDraw(Canvas canvas) {
        int w = this.getWidth();
        int h = this.getHeight();
        rect.left = 0;
        rect.top = 0;
        rect.right = w;
        rect.bottom = h;

        path.addRoundRect(rect, radius, radius, Path.Direction.CW);
        canvas.clipPath(path);
        super.onDraw(canvas);
    }
}

File: base/src/main/java/com/widget/RoundRectCornerImageView.kt
package com.widget

import android.content.Context
import android.graphics.Canvas
import android.graphics.Path
import android.graphics.RectF
import android.util.AttributeSet
import com.base.R

class RoundRectCornerImageView : android.support.v7.widget.AppCompatImageView {

    private var radius = 5.0f
    private var path: Path? = null
    private var rect: RectF? = null

    constructor(context: Context) : super(context) {
        init(context, null)
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init(context, attrs)
    }

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle) {
        init(context, attrs)
    }

    private fun init(context: Context, attrs: AttributeSet?) {
        path = Path()
        rect = RectF()
        attrs?.let {
            val typedArray = context.obtainStyledAttributes(it, R.styleable.RoundRectCornerImageView)
            radius = typedArray.getFloat(R.styleable.RoundRectCornerImageView_radiusRound, 5.0f)
            typedArray.recycle()
        }
    }

    override fun onDraw(canvas: Canvas) {
        val w = this.width
        val h = this.height

        rect?.let {
            it.left = 0f
            it.top = 0f
            it.right = w.toFloat()
            it.bottom = h.toFloat()
        }

        path?.let {
            it.addRoundRect(rect, radius, radius, Path.Direction.CW)
            canvas.clipPath(it)
        }
        super.onDraw(canvas)
    }
}
